jfix-zookeeper/src/main/java/ru/fix/zookeeper/lock/PersistentExpiringLockManager.java

```diff
@@ -180,4 +153,38 @@ public void close() {
         locks.clear();
         lockProlongationScheduler.close();
     }
+
+    /**
+     * Try to prolong lock or else remove it from {@link #locks}
+     */
+    private void prolongOrRemoveLock(LockIdentity lockId, LockContainer lockContainer) {
+        boolean prolonged = false;
+        try {
```

Comment:
If I'm not mistaken, we're trying to check if lock still exists and prolong it, and then by exception thrown are trying to guess wherever it was already removed. For me it looks kind of dirty. 
If we really want to ignore the fact that lock was already closed during the regular scheduled prolongation (do we?), then we should add to PersistentExpiringDistributedLock method which will do prolongation without throwing exception in case lock is already closed.

Reply:
>and then by exception thrown are trying to guess wherever it was already removed

`lock.checkAndProlongIfExpiresIn`  returns false if node already removed, it should not throw an exception in this case

>then we should add to PersistentExpiringDistributedLock method which will do prolongation without throwing exception in case lock is already closed

It already behaves this way, take a look at: 
`killing active lock node in zk leads to failed release` test case

Reply:
In the metioned test case you emulate the "closed" situation by straight removing value from zk:
```
zkServer.client.delete().forPath(lockPath(id))
```
therefore, this `lock`'s `.isClosed()` will return false and `assertNotClosed()` will not raise an exception. As it seems to me, we have different problem - `PersistentExpiringLockManager.close()` was called, but while its working `prolongOrRemoveLock` was also called for some `lock` which is being closed. When this `lock.close` completes - starts the body of `lock.checkAndProlongIfExpiresIn(...)`, which first line `assertNotClosed();` raises an exception.

But! Without clear test which fails without code changes and works well with new version this remains only suggestion. I keep insisting on adding clear test case on this behavior.